substrate/frame/executive/src/tests.rs

```diff
@@ -0,0 +1,1297 @@
+		type Call = Call<T>;
+
+		type Error = sp_inherents::MakeFatalError<()>;
+
+		const INHERENT_IDENTIFIER: [u8; 8] = *b"test1235";
+
+		fn create_inherent(_data: &InherentData) -> Option<Self::Call> {
+			None
+		}
+
+		fn is_inherent(call: &Self::Call) -> bool {
+			*call == Call::<T>::inherent {}
+		}
+	}
+
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+		type Call = Call<T>;
+
+		// Inherent call is accepted for being dispatched
+		fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
+			match call {
+				Call::allowed_unsigned { .. } => Ok(()),
+				Call::inherent { .. } => Ok(()),
+				_ => Err(UnknownTransaction::NoUnsignedValidator.into()),
+			}
+		}
+
+		// Inherent call is not validated as unsigned
+		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+			match call {
+				Call::allowed_unsigned { .. } => Ok(Default::default()),
+				_ => UnknownTransaction::NoUnsignedValidator.into(),
+			}
+		}
+	}
+}
+
+frame_support::construct_runtime!(
+	pub struct Runtime
+	{
+		System: frame_system::{Pallet, Call, Config<T>, Storage, Event<T>},
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		TransactionPayment: pallet_transaction_payment::{Pallet, Storage, Event<T>},
+		Custom: custom::{Pallet, Call, ValidateUnsigned, Inherent},
+		Custom2: custom2::{Pallet, Call, ValidateUnsigned, Inherent},
+	}
+);
+
+parameter_types! {
+	pub BlockWeights: frame_system::limits::BlockWeights =
+		frame_system::limits::BlockWeights::builder()
+			.base_block(Weight::from_parts(10, 0))
+			.for_class(DispatchClass::all(), |weights| weights.base_extrinsic = Weight::from_parts(5, 0))
+			.for_class(DispatchClass::non_mandatory(), |weights| weights.max_total = Weight::from_parts(1024, u64::MAX).into())
+			.build_or_panic();
+	pub const DbWeight: RuntimeDbWeight = RuntimeDbWeight {
+		read: 10,
+		write: 100,
+	};
+}
+impl frame_system::Config for Runtime {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type BlockWeights = BlockWeights;
+	type BlockLength = ();
+	type DbWeight = ();
+	type RuntimeOrigin = RuntimeOrigin;
+	type Nonce = u64;
+	type RuntimeCall = RuntimeCall;
+	type Hash = sp_core::H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = u64;
+	type Lookup = IdentityLookup<u64>;
+	type Block = TestBlock;
+	type RuntimeEvent = RuntimeEvent;
+	type BlockHashCount = ConstU64<250>;
+	type Version = RuntimeVersion;
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+	type MaxConsumers = ConstU32<16>;
+	type SingleBlockMigrations = ();
+	type MultiBlockMigrator = ();
+	type PreInherents = ();
+	type PostInherents = ();
+	type PostTransactions = ();
+}
+
+type Balance = u64;
+impl pallet_balances::Config for Runtime {
```

Comment:
and here?

Reply:
Oh, i did not know that we also have it here, yes nice.